Common/src/Pollster/Pollster.cs

```diff
@@ -251,40 +270,56 @@ void RecordError(Exception e)
+            taskHandlerLogger.LogDebug("Start a new Task to process the messageHandler");
 
-            try
+            while (runningTaskQueue_.RemoveException(out var exception))
             {
-              await using var taskHandler = new TaskHandler(sessionTable_,
-                                                            taskTable_,
-                                                            resultTable_,
-                                                            submitter_,
-                                                            dataPrefetcher_,
-                                                            workerStreamHandler_,
-                                                            message,
-                                                            taskProcessingChecker_,
-                                                            ownerPodId_,
-                                                            ownerPodName_,
-                                                            activitySource_,
-                                                            agentHandler_,
-                                                            logger_,
-                                                            pollsterOptions_,
-                                                            cts);
-
-              StopCancelledTask = taskHandler.StopCancelledTask;
+              if (exception is RpcException rpcException && TaskHandler.IsStatusFatal(rpcException.StatusCode))
+              {
+                // This exception should stop pollster
+                Console.WriteLine(exception);
+                ExceptionDispatchInfo.Capture(exception)
+                                     .Throw();
+              }
+
+              RecordError(exception);
+            }
+
+            var taskHandler = new TaskHandler(sessionTable_,
+                                              taskTable_,
+                                              resultTable_,
+                                              submitter_,
+                                              dataPrefetcher_,
+                                              workerStreamHandler_,
+                                              message,
+                                              taskProcessingChecker_,
+                                              ownerPodId_,
+                                              ownerPodName_,
+                                              activitySource_,
+                                              agentHandler_,
+                                              taskHandlerLogger,
+                                              pollsterOptions_,
+                                              () => taskProcessingDict_.Remove(message.TaskId),
+                                              cts);
+
+            taskProcessingDict_.TryAdd(message.TaskId,
+                                       taskHandler);
```

Comment:
check if add was successfull. If id was already in the dictionary, current pod is already processing this very task.